src/tcp.rs

```diff
@@ -1,19 +1,28 @@
-use crate::{Authentication, Error, IntoTargetAddr, Result, TargetAddr, ToProxyAddrs};
 use bytes::{Buf, BufMut};
+use crate::{Authentication, Error, IntoTargetAddr, TargetAddr, ToProxyAddrs};
 use derefable::Derefable;
-use futures::{stream, try_ready, Async, Future, Poll, Stream};
+use futures::Poll;
+use futures::task::Context;
+use futures::{stream, Stream, StreamExt};
 use std::borrow::Borrow;
-use std::io::{self, Read, Write};
 use std::net::{Ipv4Addr, Ipv6Addr, SocketAddr};
-use tokio_io::{AsyncRead, AsyncWrite};
-use tokio_tcp::{ConnectFuture as TokioConnect, TcpStream};
+use std::pin::Pin;
+use tokio_io::{AsyncReadExt, AsyncWriteExt, AsyncWrite};
+use tokio_net::tcp::TcpStream;
+use futures::stream::Fuse;
+use tokio_io::AsyncRead;
+use std::io;
+use std::io::ErrorKind;
 
 #[repr(u8)]
 #[derive(Clone, Copy)]
 enum Command {
     Connect = 0x01,
     Bind = 0x02,
+    #[allow(dead_code)]
     Associate = 0x03,
+    TorResolve = 0xF0,
+    TorResolvePtr = 0xF1,
```

Comment:
Tor related features are not standard socks5. Could you make them under a feature flag?

Reply:
Good idea!